templates/contribute/git.md

```diff
@@ -0,0 +1,324 @@
+
+```bash
+git remote -v
+```
+
+You should see:
+```
+origin    https://github.com/YOUR_USERNAME/mathlib4.git (fetch)
+origin    https://github.com/YOUR_USERNAME/mathlib4.git (push)
+upstream  https://github.com/leanprover-community/mathlib4.git (fetch)
+upstream  https://github.com/leanprover-community/mathlib4.git (push)
+```
+
+## Step 4: Configure the Master Branch
+
+First, fetch the branches from upstream:
+
+```bash
+git fetch upstream
+```
+
+Then make sure your `master` branch tracks `upstream/master`:
+
+```bash
+git branch --set-upstream-to=upstream/master master
+```
+
+## Step 5: Configure Git for Your Workflow
+
+### Set Default Push Behavior
+
+Configure git to push new branches to `origin` by default:
+
+```bash
+git config push.default current
+git config push.autoSetupRemote true
+```
+
+### Configure GitHub CLI (if installed)
+
+If you have the GitHub CLI installed, set the default repository to the upstream mathlib4:
+
+```bash
+gh repo set-default leanprover-community/mathlib4
+```
+
+This ensures that GitHub CLI commands like `gh pr checkout` will work with the main mathlib4 repository rather than your fork.
+
+### Prevent Accidental Commits to Master (Optional but Recommended)
+
+To avoid accidentally committing directly to `master`, you can set up a pre-commit hook. First, create the hook file:
+
+```bash
+mkdir -p .git/hooks
+cat > .git/hooks/pre-commit << 'EOF'
+#!/bin/sh
+branch="$(git rev-parse --abbrev-ref HEAD)"
+if [ "$branch" = "master" ]; then
+  echo "You can't commit directly to master branch"
+  exit 1
+fi
+EOF
+```
+
+Then make it executable:
+
+```bash
+chmod +x .git/hooks/pre-commit
+```
+
+---
+
+# Part 2: Daily Workflow
+
+These are the operations you'll use regularly when working on contributions.
+
+## Creating and Working on a New Branch
+
+### Keep Your Master Branch Up to Date
+
+**Do this before creating a new branch** to ensure you're working with the latest changes:
+
+```bash
+git switch master
+git pull
+```
+
+### Create a New Branch
+
+Then create and switch to a new branch:
+
+```bash
+git switch -c my-feature-branch
+```
+
+The branch will automatically track `origin/my-feature-branch` when you first push it.
+
+## Push Your Branch and Open a PR
+
+### Push Your Branch
+
+After making your changes and commits:
+
+```bash
+git push origin my-feature-branch
+```
+
+### Open a Pull Request
+
+1. Go to your fork on GitHub: `https://github.com/YOUR_USERNAME/mathlib4`
+2. You should see a banner suggesting to open a PR for your recent push
+3. Click "Compare & pull request"
+4. Fill in the PR title and description
+5. Click "Create pull request"
+
+Alternatively, or if you do not see the banner, you can also go to https://github.com/leanprover-community/mathlib4/compare, and click `compare across forks`. You will need to select your fork in the "head repository" drop-down menu, and select the branch you want to merge in the "compare" drop-down menu.
+## Working with Others' PRs
```

Comment:
Here we should mention *very clearly* that even just opening VS Code will execute code written by a random person on the internet, with ZERO (*ZERO*) control by the mathlib community.

Reply:
Ah, I see now that there's some text about security in the "Additional Information" section.